controllers/horizon_controller.go

```diff
@@ -547,3 +572,52 @@ func (r *HorizonReconciler) ensureHorizonSecret(
 
 	return nil
 }
+
+func (r *HorizonReconciler) renderMemcached(instance *horizonv1alpha1.Horizon) *memcachedv1.Memcached {
+	return &memcachedv1.Memcached{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "memcached.openstack.org/v1beta1",
+			Kind:       "Memcached",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-memcached", instance.Name),
+			Namespace: instance.Namespace,
+		},
+		Spec: memcachedv1.MemcachedSpec{
+			Replicas: 3,
+		},
+	}
+}
+
+func (r *HorizonReconciler) checkMemcachedCR(ctx context.Context, instance *horizonv1alpha1.Horizon) (bool, error) {
+	list := &memcachedv1.MemcachedList{}
+	if err := r.Client.List(ctx, list); err != nil {
+		r.Log.Info("Error while getting list of memcaches")
+		return false, err
+	}
+	for _, memcached := range list.Items {
+		if memcached.ObjectMeta.Name == fmt.Sprintf("%s-memcached", instance.Name) {
```

Comment:
the instance name should be configurable so that users can use a single cluster rather than multiple ones.

Reply:
I wonder if we would want the user to specify the memcached instance. 

Or rather, just a Spec item like `sharedMemcached` that takes a bool. Then have a function that searches for existing `memcached` implementations.

I kind of prefer the latter

Reply:
I would like to make the connection between the Horizon/my-horizon instance and the Memcached/my-horizon-memcached explicit. As this is now hardcodes a connection implicitly that is pretty hard to discover from outside, looking at the CRs in the system.

I would do that by adding a MemcachedInstance field to HorizonSpec to carry the name of the Memcached CR instance the Horizon instance supposed to use.

I would make that field Optional. And then Horizon can create its own Memcached instance if value for the MemcachedInstance is not provided in the CR.

Reply:
This has been implemented in the latest patchset.